data_reader.py

```diff
@@ -0,0 +1,121 @@
+import glob
+import collections
+import os
+
+from weather import Weather
+
+
+class DataReader:
+
+	annual_report = {}
+
+	def print_annual_report(self, report_data):
+		print('-'*80)
+		print('{:<10s}{:<10s}{:>12s}{:>18s}{:>20s}'.format(
+			'Year', 'MAX Temp', 'MIN Temp', 'MAX Humidity', 'MIN Humidity'
+		))
+		print('-'*80)
+		
+		for key, value in sorted(report_data.items()):
+			print('{:<10s}{:^10s}{:^12s}{:^18s}{:^20s}'.format(
+				value.year, str(value.max_temp), str(value.min_temp), str(value.max_humidity), str(value.min_humidity)
+			))
+
+	def print_annual_max_report(self, report_data):
+		print('-'*40)
+		print('{:<10s}{:^10s}{:>10s}'.format('Year', 'Date', 'Temp'))
+		print('-'*40)
+		
+		for key, value in sorted(report_data.items()):
+			print('{:<10s}{:<10s}{:>8s}'.format(
+				value.year, value.date, str(value.max_temp)
+			))
+	
+	def set_weather_attributes(self, weather):
+		if not self.annual_report.get(weather.year):
+			self.annual_report[weather.year] = weather
+			return
+
+		if weather.max_temp > self.annual_report[weather.year].max_temp or self.annual_report[weather.year].max_temp == None: 
+			self.annual_report[weather.year].date = weather.date
+			self.annual_report[weather.year].max_temp = weather.max_temp
+
+		if weather.min_temp < self.annual_report[weather.year].min_temp or self.annual_report[weather.year].min_temp == None:
+			self.annual_report[weather.year].min_temp = weather.min_temp
+		
+		if weather.max_humidity > self.annual_report[weather.year].max_humidity or self.annual_report[weather.year].max_humidity == None:
+			self.annual_report[weather.year].max_humidity = weather.max_humidity
+		
+		if weather.min_humidity < self.annual_report[weather.year].min_humidity or self.annual_report[weather.year].min_humidity == None:
+			self.annual_report[weather.year].min_humidity = weather.min_humidity
+
+	def extract_files(self, dir_path):
+
+		if os.listdir(dir_path):
+			for item in os.listdir(dir_path):
+
+				if os.path.isfile(os.path.join(dir_path,item)):
+					self.read_file(os.path.join(dir_path,item))
+		else:
+			print 'No files were found!'
+			return
+
+	def generate_annual_report(self, dir_path):
+
+		self.extract_files(dir_path)
+		return self.annual_report
+	
+	def filter_each_line(self, item):
+
+		for line in item:
+
+			if not line.isspace() or line.startswith('<!--'):
+				yield line		
+
+	def read_file(self, textfile):
+
+		try: 
+			with open(textfile) as item:
+
+				filtered_file = self.filter_each_line(item)
+				file_reader = csv.DictReader(filtered_file, delimiter=',')
+				yearly_report = self.manipulate_attributes(file_reader, textfile.split('_')[2])
+				
+			return yearly_report
+
+		except IOError:
+			print 'Could not read file: ', textfile
+
+		except:
+			raise
+
+	def manipulate_attributes(self, file_reader, filename):
+		max_temp = None
+		min_temp = None
+		max_humidity = None
+		min_humidity = None
+		max_date = ''
+
+		for row_data in file_reader:
+
+			if 'PKT' in row_data:
```

Comment:
can use
```
my_dict.get('PKT', None)
```
to avoid this condition and similar could be used in following statements